pyproject.toml

```diff
@@ -84,3 +84,5 @@ version_files = [
     "pyproject.toml:^version"
 ]
 
+[tool.mypy]
+plugins = "pydantic.mypy"
```

Comment:
This "fixes" this error:
![image](https://github.com/BerriAI/litellm/assets/116613820/b59a4025-2852-4477-aa0a-42cc7bf84d0d)

mypy treats `model_config` assignment as an unintended override -- when it is.
Adding the pydantic mypy plugin seems to solve this problem. https://docs.pydantic.dev/latest/integrations/mypy/

After adding this:
mypy . --ignore-missing-imports | grep "Cannot override class variable (previously declared on base class "BaseModel") with instance variable" return zero lint errors relating to overring class variaables.